pkg/gts/gts_status.go

```diff
@@ -0,0 +1,62 @@
+package gts
+
+import (
+	"bytes"
+	"crypto/x509"
+	"fmt"
+	"github.com/samber/lo"
+	"github.com/wttech/aemc/pkg/common/fmtx"
+	"io"
+)
+
+type Status struct {
+	Created      bool          `json:"exists"`
+	Certificates []Certificate `json:"aliases"`
+}
+
+func (s *Status) FindCertificate(certificate x509.Certificate) (*Certificate, error) {
+	for i := range s.Certificates {
+		isEqualResult, err := s.Certificates[i].IsEqual(certificate)
+
+		if err != nil {
+			return nil, err
+		}
+
+		if isEqualResult {
+			return &s.Certificates[i], nil
+		}
+	}
+
+	return nil, nil
+}
+
+func (s *Status) FindCertificateByAlias(alias string) *Certificate {
+	for i := range s.Certificates {
+		if s.Certificates[i].Alias == alias {
+			return &s.Certificates[i]
+		}
+	}
+
+	return nil
+}
+
+func UnmarshalStatus(readCloser io.ReadCloser) (*Status, error) {
+
```

Comment:
Format of data should never change! This is a very often cause of app errors. Format of data is a contract between apps.